### PR TITLE
feat: add warning about v8 snapshots to troubleshooting guide

### PR DESCRIPTION
**Pull request by ryanthemanuel**
Add warning about v8 snapshots to troubleshooting guide

### CODE REVIEW

File: content/guides/references/troubleshooting.md [hunk 1]
```diff
@@ -633,6 +633,15 @@ official fix faster.
 
 ### Edit the installed Cypress code
 
+<Alert type="warning">
+
+In Cypress 11.0, Cypress started using [v8 snapshots](https://v8.dev/blog/custom-startup-snapshots)
```
Comment (lmiller1990) [inline content/guides/references/troubleshooting.md:638]:
```suggestion
In Cypress 11.0, Cypress uses [v8 snapshots](https://v8.dev/blog/custom-startup-snapshots)
```

Comment (lmiller1990) [reply content/guides/references/troubleshooting.md:638]: I think it reads a little better. Good idea to warn about this.

**Review by lmiller1990 — APPROVED** [review]

File: content/guides/references/troubleshooting.md [hunk 2]
```diff
@@ -633,6 +633,17 @@ official fix faster.
...
+time. As a result, much of the Cypress code is contained in a single
+unmodifiable snapshot file. Thus, editing the Cypress code in the binary
+directly as described below may or may not work depending on where and how that
+code is executed.
+
+</Alert>
+
 The installed application comes with the fully transpiled, unobfuscated
 JavaScript source code that you can hack on. You might want to directly modify
 the installed app code to:
```
Comment (chrisbreiding) [inline content/guides/references/troubleshooting.md:649]: With V8 snapshots, "fully transpiled, unobfuscated JavaScript source code" isn't true, right? I think we should remove that language and anything below that doesn't apply with snapshots. The docs only cover the current version of Cypress, so we should target this towards the `release-11.0` branch and the verbiage should be more clear about what can and can't be modified by users.

**Review by chrisbreiding — CHANGES_REQUESTED** [review]

File: content/guides/references/troubleshooting.md (hunk 2 above)
Comment (ryanthemanuel) [reply content/guides/references/troubleshooting.md:649]: Ah gotcha. Yeah I thought that we potentially tried to cover multiple versions, but I can update this to be specific to 11.0

Comment (ryanthemanuel) [reply content/guides/references/troubleshooting.md]: Good call. Thinking about this more, I think this section will cause more confusion than any benefit with leaving it. I decided to remove it instead:
[`dfbab20` (#4825)](https://github.com/cypress-io/cypress-documentation/pull/4825/commits/dfbab207630f5c6934cde44ce6f943a63a91ddca)

File: content/guides/references/troubleshooting.md [hunk 3]
```diff
@@ -630,42 +630,3 @@ workaround to the relevant Cypress GitHub issue. It will help us release an
...
-Second, open the source code at the following path in any code editor. Make sure
-to substitute `3.3.1` for the desired version of Cypress you want to edit.
-
-```text
-/Users/jane/Library/Caches/Cypress/3.3.1/Cypress.app/Contents/Resources/app/packages/
-```
-
-You can change anything in the JavaScript code:
-
-<DocsImage src="/img/guides/troubleshooting/source-code.png" alt="Source code of Cypress in a text editor" ></DocsImage>
```
Comment (chrisbreiding) [inline content/guides/references/troubleshooting.md:663]: I think we can remove this image as well. It doesn't seem to be referenced anywhere else.

Comment (ryanthemanuel) [reply content/guides/references/troubleshooting.md:663]: Good catch.

Comment (ryanthemanuel) [reply content/guides/references/troubleshooting.md:663]: Removed: [`92e4ed8` (#4825)](https://github.com/cypress-io/cypress-documentation/pull/4825/commits/92e4ed8b44eecee78751c7cad748a1ef8814f626)

**Review by chrisbreiding — APPROVED** [review]

**Pull request merged**